cosmic-settings/src/pages/power/backend/mod.rs

```diff
@@ -270,81 +315,149 @@ async fn get_on_battery_status() -> Result<bool, zbus::Error> {
 }
 
 impl Battery {
-    pub async fn update_battery() -> Self {
-        let proxy = get_device_proxy().await;
+    pub async fn from_device(proxy: DeviceProxy<'_>) -> Self {
```

Comment:
Ideally, I would like to implement the From trait, but I have no idea how to do so with an async function. Any suggestion? Should I keep it like this?

Reply:
You cannot do that since it's not an async trait method.